patroni/postgresql/postmaster.py

```diff
@@ -113,10 +114,18 @@ def signal_stop(self, mode, pg_ctl='pg_ctl'):
         if self.is_single_user:
             logger.warning("Cannot stop server; single-user server is running (PID: {0})".format(self.pid))
             return False
-        if os.name != 'posix':
+        if os.name != 'posix' and pg_ctl:
             return self.pg_ctl_kill(mode, pg_ctl)
         try:
+            children = self.children()
             self.send_signal(getattr(signal, 'SIG' + STOP_SIGNALS[mode]))
+            if mode == 'abort':
+                _, alive = psutil.wait_procs(children, timeout=1)
+                for child in alive:
+                    try:
+                        child.send_signal(getattr(signal, 'SIG' + STOP_SIGNALS[mode]))
+                    except psutil.NoSuchProcess:
+                        pass
```

Comment:
I think it would be better to extract handling of `abort` into a separate method.
This is a very special operation and we don't even need to use the pg_ctl for that. Essentially we want to hardly kill postmaster and all ancestors.
In order to make sure that no new child processes are created by postmaster I would also `suspend` it. We already doing something similar when [terminating](https://github.com/zalando/patroni/blob/master/patroni/postgresql/cancellable.py#L34-L52) old callbacks

Reply:
Thankyou. I will update the code and the unit tests.